### DISCUSSION TITLE
Handling Action Failure - how do you do it ?

### DISCUSSION THREAD

**Discussion by poirazis** [Budibase/budibase#7129] [General]
I have a form to edit a contact entity, where a column "mobPhone" has a UNIQUE constrain on it back in MySQL
during editing a user may enter a non unique mobPhone, and the Save action of the Form will fail with an error message.

how would you / do you handle these cases ?

the "dream" result would have been for the field that violated the rule to show it's validation error message and get the focus.

**Comment by melohagan** [comment]
Hey @poirazis 

The way I would handle this would be to have a "lookup" button that calls a custom query to check if the number exists. 
You could then save this SQL COUNT to app state.

If the count is > 0, then you could display a conditional UI red paragraph that states the number is already in use - try again. 

I know it's a bit of hassle, but I would argue that what you're trying to do is not validation, it's **verification**. The user could be entering a valid phone number - you're verifying that number is unique.
